### ISSUE TITLE
Implement Autotools and Ratio Groups

### ISSUE THREAD

**Issue by Porco-Rosso** [jfurrow/Flood#241]
Hi, 

Firstly, thanks for implementing my last suggestion.
Personally, what is keeping me from adopting flood as my main rtorrent interface is the lack of the features offered by these two plugins:

- [Autotools](https://github.com/Novik/ruTorrent/wiki/PluginAutotools)
- [Ratio Groups ](https://github.com/rakshasa/rtorrent/wiki/RTorrentRatioHandling)

Based on my experience, these are features that are very useful for power users. The first allows automation, while the second allows to efficiently deal with both private and public trackers. 
If you are considering further features, these would be useful. 
Thanks

**Comment by DavePS** [comment]
Yes `Autotools` is more than usefull, by the way this is cool that we can choose the correct download folder before starting to download.

**Comment by jfurrow** [comment]
Agreed!

**Comment by madcatsu** [comment]
With regards to Autotools functionality, some ideas to play with might be:

- Actions should be able to pass metadata such as:
- + torrent name
- + torrent path
- + rtorrent's configured download folder
- + tracker dns name OR ip (whichever is available from the original .torrent file or magnet link)
- + label
- Actions possible could be a post finish event which might:
- + relabel torrent
- + change tracker announce address
- + call external executable or script with any of the above metadata as variables
- + move, copy or hardlink completed torrents to another folder location 

Today this can be achieved with a heavily customized rtorrent.rc file, but I think's Flood's fantastic UI should abstract these customizations and handle these events through the XMLRPC interface
